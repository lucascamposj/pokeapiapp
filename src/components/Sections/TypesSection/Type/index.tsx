import React from 'react';
import TypesObject from './types';
import { TypeImage, Container } from './styles';
import { TypeInterface } from '../../../../pages/Pokemon';

const Type: React.FC<TypeInterface> = ({ type }: TypeInterface) => {
  const { name } = type;
  const source = TypesObject[name]?.image;
  console.log(name);
  console.log(source);
  return (
    <>
      {source && (
        <Container>
          <TypeImage source={TypesObject[name].image} resizeMode="contain" />
        </Container>
      )}
    </>
  );
};

export default Type;
