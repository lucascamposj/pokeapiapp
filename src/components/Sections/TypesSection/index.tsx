import React from 'react';
import Type from './Type';
import { Container } from './styles';
import { TypeInterface } from '../../../pages/Pokemon';

interface TypeSectionProps {
  types: TypeInterface[] | undefined;
}

const TypeSection: React.FC<TypeSectionProps> = ({
  types,
}: TypeSectionProps) => {
  return (
    <Container>
      {types?.map(({ type, slot }) => (
        <Type key={type.name} type={type} slot={slot} />
      ))}
    </Container>
  );
};

export default TypeSection;
