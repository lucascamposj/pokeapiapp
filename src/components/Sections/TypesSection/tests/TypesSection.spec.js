import React from 'react';
import {shallow} from 'enzyme';
import TypesSection from '../../TypesSection';

describe('TypesSection', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const types = [
              {
                slot: 3,
                type: {
                  name: "electric",
                  url: "electric.com"
                }
              },
              {
                slot: 2,
                type: {
                  name: "poison",
                  url: "electric.com"
                }
              }
            ]
            const component = shallow(
              <TypesSection
                types={types}
              />)
            expect(component).toMatchSnapshot()
        });
    });
});
