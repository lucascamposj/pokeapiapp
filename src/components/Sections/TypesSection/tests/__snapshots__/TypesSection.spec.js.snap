// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypesSection Rendering should match to snapshot 1`] = `
<Styled(View)>
  <Type
    key="electric"
    slot={3}
    type={
      Object {
        "name": "electric",
        "url": "electric.com",
      }
    }
  />
  <Type
    key="poison"
    slot={2}
    type={
      Object {
        "name": "poison",
        "url": "electric.com",
      }
    }
  />
</Styled(View)>
`;
